new_client/src/auth/Signup.jsx

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import { Container, Paper, Grid, TextField, Button } from '@material-ui/core';
+import "./Error.css";
+
+export default function Signup() {
+	const [form, updateForm] = React.useState({
+		username: "",
+		email: "",
+		password: "",
+		graduationYear: null,
+	});
+
+	const [errors, updateErrors] = React.useState({
+		username: "",
+		email: "",
+		password: "",
+		graduationYear: null
+	});
+
+	function isFormValid() {
+		let formIsValid = true;
+		if (!form.username) {
+			formIsValid = false;
+			updateErrors(prevErrors => ({
+				...prevErrors,
+				username: "*Username can't be Empty"
+			}));
+		}
+
+		if (!form.email) {
+			formIsValid = false;
+			updateErrors(prevErrors => ({
+				...prevErrors,
+				email: "*Email can't be Empty",
+			}));
+		}
+
+		if (typeof form.email !== undefined) {
+			var pattern = new RegExp(/^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i);
+			if (!pattern.test(form.email)) {
+				formIsValid = false;
+				updateErrors(prevErrors => ({
+					...prevErrors,
+					errors: "*Please enter valid Email ID",
+				}));
+			}
+		}
+
+		if (!form.password) {
+			formIsValid = false;
+			updateErrors(prevErrors => ({
+				...prevErrors,
+				password: "*Please enter your password."
+			}));
+		}
+
+		// Password Constraints relaxed for now.
```

Comment:
as and when free please add the contsraints ....although you did mention in the comments

Reply:
Will add them

Reply:
resolved